packages/wonder-blocks-dropdown/components/dropdown-core.js

```diff
@@ -117,91 +113,96 @@ export default class DropdownCore extends React.Component<DropdownCoreProps> {
+        if (openerElement) {
+            const rect = openerElement.getBoundingClientRect();
+            return {
                 top: rect.top + window.pageYOffset,
                 left: rect.left + window.pageXOffset,
                 width: rect.width,
             };
         }
-    };
+    }
 
-    findTranslatedPosition() {
-        if (this.openerBounds) {
-            const {top, left, width} = this.openerBounds;
-            if (this.props.alignment === "left") {
-                return {
-                    transform: `translate3d(${left}px, ${top}px, 0)`,
-                };
-            } else {
-                return {
-                    transform: `translate3d(${left +
-                        width -
-                        (this.menuWidth || 0)}px, ${top}px, 0)`,
-                };
-            }
+    getTranslatedPosition(openerBounds: {
+        top: number,
+        left: number,
+        width: number,
+    }) {
+        const {top, left, width} = openerBounds;
+        if (this.props.alignment === "left") {
+            return {
+                transform: `translate3d(${left}px, ${top}px, 0)`,
+            };
+        } else {
+            return {
+                transform: `translate3d(${left +
+                    width -
+                    (this.menuWidth || 0)}px, ${top}px, 0)`,
+            };
         }
     }
 
     maybeRenderMenu() {
         const {items, light, open, style} = this.props;
+        const body = document.getElementsByTagName("body");
+
+        const bounds = this.getOpenerBounds();
+        const translatedPosition = bounds && this.getTranslatedPosition(bounds);
 
-        this.updatePosition();
-        if (open) {
-            return (
+        if (open && body[0]) {
+            return ReactDOM.createPortal(
                 <View
-                    onClick={(event) => {
-                        // Stop propagation to prevent the mouse click listener on the
-                        // document from closing the menu.
+                    onMouseUp={(event) => {
+                        // Stop propagation to prevent the mouseup listener
+                        // on the document from closing the menu.
                         event.nativeEvent.stopImmediatePropagation();
                     }}
-                    ref={(node) => (this.menuRef = ReactDOM.findDOMNode(node))}
+                    ref={(node) =>
+                        (this.menuElement = ((ReactDOM.findDOMNode(
+                            node,
+                        ): any): Element))
+                    }
                     style={[
                         styles.dropdown,
                         light && styles.light,
-                        (!open || !this.menuWidth) && styles.hide,
-                        this.findTranslatedPosition(),
+                        !(open && this.menuWidth) && styles.hide,
+                        translatedPosition,
                         style,
```

Comment:
I'm not sure this is the element we want to be apply the `style`.  The purpose of providing a `style` element on things to be able to adding `margin` for spacing or `width`, `align-self`, etc. for sizing.

Reply:
Also, one of the examples should show `style` in action.

Reply:
This is one of the things I'm going to address with the refactoring, because that would allow users to apply styles more intuitively. I'll add a TODO.